sqlite.go

```diff
@@ -823,41 +928,37 @@ func (s *stmt) bind(stmt uintptr, n int, args []driver.Value) error {
 		return fmt.Errorf("missing arguments: got %v, expected %v", len(args), n)
 	}
 
+	var err error
+
 	for i, v := range args[:n] {
 		switch x := v.(type) {
+		case nil:
+			err = s.bindNull(stmt, i+1)
 		case int64:
-			if err := s.bindInt64(stmt, i+1, x); err != nil {
-				return err
-			}
+			err = s.bindInt64(stmt, i+1, x)
 		case float64:
-			if err := s.bindDouble(stmt, i+1, x); err != nil {
-				return err
-			}
+			err = s.bindDouble(stmt, i+1, x)
 		case bool:
 			v := 0
 			if x {
 				v = 1
 			}
-			if err := s.bindInt(stmt, i+1, v); err != nil {
-				return err
-			}
+			err = s.bindInt(stmt, i+1, v)
 		case []byte:
-			if err := s.bindBlob(stmt, i+1, x); err != nil {
-				return err
-			}
+			err = s.bindBlob(stmt, i+1, x)
 		case string:
-			if err := s.bindText(stmt, i+1, x); err != nil {
-				return err
-			}
+			err = s.bindText(stmt, i+1, x)
 		case time.Time:
-			if err := s.bindText(stmt, i+1, x.String()); err != nil {
-				return err
+			if s.conn.compatibility == "mattn" {
+				err = s.bindText(stmt, i+1, x.Format(MattnTimestampFormats[0]))
+			} else {
+				err = s.bindText(stmt, i+1, x.String())
 			}
 		default:
 			return fmt.Errorf("invalid driver.Value type %T", x)
 		}
 	}
-	return nil
+	return err
```

Comment:
If len(args) > 1 and binding any of the arguments except the last one returns an error the error will be silently ignored.